addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import logging
+import requests
+from werkzeug.exceptions import Forbidden
+
+from odoo import http, tools, _
+from odoo.addons.iap.tools import iap_tools
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class MailPluginController(http.Controller):
+
+    @http.route('/mail_client_extension/modules/get', type="json", auth="outlook", csrf=False, cors="*")
+    def modules_get(self, **kwargs):
+        """
+            deprecated as of saas-14.3, not needed for newer versions of the mail plugin but necessary
+            for supporting older versions
+        """
+        return {'modules': ['contacts', 'crm']}
+
+    @http.route('/mail_plugin/partner/enrich_and_create_company',
+                type="json", auth="outlook", cors="*")
+    def res_partner_enrich_and_create_company(self, partner_id):
+        """
+        Route used when the user clicks on the create and enrich partner button
+        it will try to find a company using IAP, if a company is found
+        the enriched company will then be created in the database
+        """
+        response = {}
+
+        partner = request.env['res.partner'].browse(partner_id)
+
+        normalized_email = partner.email_normalized
+        if not normalized_email:
+            response = {'error': 'contact has no email'}
```

Comment:
Strange, either return a key, either a translated string but seems a mix of both ?

Reply:
As explained above we now return a translated string.

Reply:
If it supports strings, could probably be "Contact has no valid email." or "Contact email is invalid."

Reply:
Okay, changed to "Contact has no valid email".